ably/proto_message.go

```diff
@@ -193,6 +241,116 @@ func (m Message) withDecodedData(cipher channelCipher) (Message, error) {
 	return m, nil
 }
 
+// withDecodedDataAndContext decodes message data with support for delta decoding (RTL19).
+func (m Message) withDecodedDataAndContext(cipher channelCipher, ctx *DecodingContext) (Message, []byte, error) {
+	// strings.Split on empty string returns []string{""}
+	if m.Data == nil || m.Encoding == "" {
+		return m, nil, nil
+	}
+
+	var basePayloadForReturn []byte
+	encodings := strings.Split(m.Encoding, "/")
+
+	for len(encodings) > 0 {
+		encoding := encodings[len(encodings)-1]
+		encodings = encodings[:len(encodings)-1]
+		switch encoding {
+		case encVCDiff:
+			// Handle vcdiff delta decoding (PC3, RTL18, RTL19, RTL20)
+			if ctx == nil || ctx.VCDiffPlugin == nil {
+				return m, nil, fmt.Errorf("missing VCdiff decoder plugin (code 40019)")
```

Comment:
Should this continue, perhaps with a log message, rather than returning an error?

Reply:
I think it shouldn't, when decoder error happens we stop processing messages and initiate reattach instead